crates/matching-engine/src/cfmm/uniswap/pool_manager.rs

```diff
@@ -46,7 +46,7 @@ where
     latest_synced_block: u64,
     state_change_buffer: usize,
     state_change_cache:  Arc<RwLock<StateChangeCache<Loader, A>>>,
-    provider:            Arc<P>,
+    pub provider:        Arc<P>,
     sync_started:        AtomicBool
```

Comment:
Why touch internals?

Reply:
needs to be cleaned up, i don't anymore. I found a bug in some of the math i was doing so fixing that now